pkg/cbox/publicshare/sql/sql.go

```diff
@@ -455,3 +468,18 @@ func checkPasswordHash(password, hash string) bool {
 	err := bcrypt.CompareHashAndPassword([]byte(strings.TrimPrefix(hash, "1|")), []byte(password))
 	return err == nil
 }
+
+func authenticate(share *link.PublicShare, pw string, auth publicshare.Authentication) bool {
+	switch {
+	case auth.Password != "":
+		return checkPasswordHash(auth.Password, pw)
+	case auth.Signature != publicshare.Signature{}:
```

Comment:
Are you sure this condition isn't true when auth.Signature is nil?

Reply:
`auth.Signature` can't be nil.